Connect authentication service to notification service

iplonk3 opened the pull request.

Commit:
Connect authentication service to notification service

Commit:
Fix merge conflicts

Commit:
Minor changes to docker compose files

Commit:
Add rabbitmq to docker compose for tests

Commit:
Provide routing key and check for login event

dmssargent commented on build.gradle:
```diff
@@ -40,9 +40,10 @@ ext {
+    compile('org.springframework.boot:spring-boot-starter-data-jpa')
     compile('org.springframework.boot:spring-boot-starter-web')
     compile('org.springframework.boot:spring-boot-starter-actuator')
+	compile('org.springframework.boot:spring-boot-starter-amqp')
```
The whitespacing is off

dmssargent commented on docker-compose.yml:
```diff
@@ -33,6 +33,15 @@ services:
+    environment:
+      - RABBIT_HOST=rabbitmq
+
+
```
there is an extra blank line here

Commit:
Distinguish between authentication success events

Commit:
Remove unused import

Commit:
Fix spacing in build.gradle

dmssargent commented on src/main/java/org/aitesting/microservices/authentication/rabbitmq/LoginSuccessListener.java:
```diff
@@ -0,0 +1,28 @@
+        if (evt.getSource() instanceof UsernamePasswordAuthenticationToken) {
+            User user = (User) evt.getAuthentication().getPrincipal();
+            log.info(String.format("User %s logged in", user.getUsername()));
+            producer.send(String.format("{\"userId\":\"%s\"}", user.getUsername()));
```
You should be using the current user id, not the user name.

Commit:
Correct the username to user id, and add the notification service to docker-compose-local.yml

jacobkissel commented on src/main/java/org/aitesting/microservices/authentication/rabbitmq/Producer.java:
```diff
@@ -0,0 +1,25 @@
+
+    public void send(String message) {
+        log.info(String.format("Sending message %s to %s", message, topicExchange.getName()));
+        this.rabbitTemplate.convertAndSend(topicExchange.getName(), "notification.user.authentication", message);
```
Routing key should be like this: {bounded-context}.{aggregate-name}.{what-happened-in-past-tense}
So like notification.user.userauthenticated or something maybe

jusoto committed:
Replaced userdetails.User for userdetails.UserDetails in LoginSuccessListener to avoid confusion between model.User and uderdetails.User.
Updated test to use SpringRunner.class, defined random port and single declaration of hostUrl.
In WebSecurityConfigurer disabled http.headers to solve issue with front-end

Commit:
Remove CORS filter

jusoto committed:
Added proper write/read of UUID to/from MySQL VARCHAR

jusoto committed:
Removed headers for CORS filter due to duplicated response headers on front-end due to edge-service is also adding these headers.
Added a try catch for exceptions when sending the message to notification service

Commit:
Fixed merge conflicts

Commit:
Change routing key and generate random uuid for users
